### PR TITLE
:sparkles: TK-3425 Add CLI support to get Application YAMLs from backup

### PR DESCRIPTION
**Pull request by vishutrilio**
Changes Done:
* Added resource-metadata CLI support for target-browser plugin

### CODE REVIEW

File: cmd/target-browser/cmd/constant.go [hunk 1]
```diff
@@ -78,11 +79,31 @@ const (
...
+	versionDefault = ""
+	versionUsage   = "version string to get all data related to API version"
+
+	kindFlag    = "kind"
+	kindDefault = ""
+	kindUsage   = "kind string to get all data related to API kind"
+
+	nameFlag    = "name"
+	nameDefault = ""
+	nameUsage   = "name string to get all data related to name"
```
Comment (msurudwad) [inline cmd/target-browser/cmd/constant.go:97]: `name of backed up resource whose resource-metadata needs to be retrived`

File: cmd/target-browser/cmd/resource-metadata.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+func init() {
+	getCmd.AddCommand(resourceMetadataCmd())
+}
+
+// resourceMetadataCmd represents the resource-metadata command
+func resourceMetadataCmd() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   ResourceMetadataCmdName,
+		Short: "Get Backup high level metadata details",
+		Long:  `Performs GET operation on target-browser's '/resource-metadata' API and gets Application YAMLs from backup in JSON format`,
```
Comment (msurudwad) [inline cmd/target-browser/cmd/resource-metadata.go:19]: `YAMLs` is wrong

File: cmd/target-browser/cmd/resource-metadata.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+`,
+		RunE: getResourceMetadata,
+	}
+
+	cmd.Flags().StringVar(&group, groupFlag, groupDefault, groupUsage)
+
+	cmd.Flags().StringVar(&backupPlanUID, BackupPlanUIDFlag, backupPlanUIDDefault, backupPlanUIDUsage)
+	err := cmd.MarkFlagRequired(BackupPlanUIDFlag)
+	if err != nil {
+		log.Fatalf("Invalid option or missing required flag %s - %s", BackupPlanUIDFlag, err.Error())
```
Comment (msurudwad) [inline cmd/target-browser/cmd/resource-metadata.go:32]: no need to mark this flag required.. use default value empty string.

File: tools/target-browser/backup.go [hunk 4]
```diff
@@ -83,7 +83,7 @@ func (auth *AuthInfo) TriggerAPI(pathParam, queryParam, apiPath string, selector
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK || resp.Body == nil {
+	if resp.StatusCode != http.StatusOK && resp.Body == nil {
```
Comment (msurudwad) [inline tools/target-browser/backup.go:86]: why this change?
Just test these changes once, if  `StatusCode` is OK and `resp.Body` is nil, all commands are working properly and printing of response is not giving error for all formats.

File: tools/target-browser/resource-metadata.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+// GetResourceMetadata returns metadata of backup on mounted target
+func (auth *AuthInfo) GetResourceMetadata(options *ResourceMetadataListOptions) error {
+	values, err := query.Values(options)
+	if err != nil {
+		return err
+	}
+	queryParam := values.Encode()
+	resp, apiErr := auth.TriggerAPI("", queryParam, internal.ResourceMetadataAPIPath, []string{})
+	if apiErr != nil {
+		log.Info(apiErr)
```
Comment (msurudwad) [inline tools/target-browser/resource-metadata.go:30]: remove this log

**Review by msurudwad — CHANGES_REQUESTED** [review]
Test-cases needs to be added for these changes.. add in this PR or create another PR

File: cmd/target-browser/cmd/constant.go [hunk 6]
```diff
@@ -78,11 +79,31 @@ const (
 	OperationScopeFlag  = "operation-scope"
 	operationScopeUsage = "Filter backup/backupPlan for [SingleNamespace, MultiNamespace]. " +
 		"Supported values can be in any case capital, small or mixed."
+
+	groupFlag    = "group"
+	groupDefault = ""
+	groupUsage   = "group string to get all data related to API group"
```
[the remark this answers is not in the document]
Comment (vishutrilio) [reply cmd/target-browser/cmd/constant.go:85]: Done

File: cmd/target-browser/cmd/constant.go [hunk 7]
```diff
@@ -78,11 +79,31 @@ const (
...
+
+	versionFlag    = "version"
+	versionDefault = ""
+	versionUsage   = "version string to get all data related to API version"
```
[the remark this answers is not in the document]
Comment (vishutrilio) [reply cmd/target-browser/cmd/constant.go:89]: Done

File: cmd/target-browser/cmd/constant.go [hunk 8]
```diff
@@ -78,11 +79,31 @@ const (
...
+
+	kindFlag    = "kind"
+	kindDefault = ""
+	kindUsage   = "kind string to get all data related to API kind"
```
[the remark this answers is not in the document]
Comment (vishutrilio) [reply cmd/target-browser/cmd/constant.go:93]: Done

File: cmd/target-browser/cmd/constant.go (hunk 1 above)
Comment (vishutrilio) [reply cmd/target-browser/cmd/constant.go:97]: Done

File: cmd/target-browser/cmd/resource-metadata.go (hunk 2 above)
Comment (vishutrilio) [reply cmd/target-browser/cmd/resource-metadata.go:19]: Changed

File: cmd/target-browser/cmd/resource-metadata.go (hunk 3 above)
Comment (vishutrilio) [reply cmd/target-browser/cmd/resource-metadata.go:32]: Kept it until server side changes are done

File: tools/target-browser/resource-metadata.go (hunk 5 above)
Comment (vishutrilio) [reply tools/target-browser/resource-metadata.go:30]: Done

File: tools/target-browser/backup.go (hunk 4 above)
Comment (vishutrilio) [reply tools/target-browser/backup.go:86]: Reverted

File: cmd/target-browser/cmd/resource-metadata.go [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+func init() {
+	getCmd.AddCommand(resourceMetadataCmd())
+}
+
+// resourceMetadataCmd represents the resource-metadata command
+func resourceMetadataCmd() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   ResourceMetadataCmdName,
+		Short: "Get Backup high level metadata details",
```
[the remark this answers is not in the document]
Comment (vishutrilio) [reply cmd/target-browser/cmd/resource-metadata.go]: Changed

**Review by msurudwad — CHANGES_REQUESTED** [review]

File: cmd/target-browser/cmd/resource-metadata.go (hunk 2 above)
Comment (vishutrilio) [reply cmd/target-browser/cmd/resource-metadata.go]: Done

File: cmd/target-browser/cmd/constant.go [hunk 10]
```diff
@@ -80,21 +80,24 @@ const (
 	operationScopeUsage = "Filter backup/backupPlan for [SingleNamespace, MultiNamespace]. " +
 		"Supported values can be in any case capital, small or mixed."
 
-	groupFlag    = "group"
-	groupDefault = ""
-	groupUsage   = "group string to get all data related to API group"
+	groupFlag      = "group"
+	groupFlagShort = "g"
+	groupDefault   = ""
+	groupUsage     = "PI group name of resource whose resource-metadata needs to be retrieved"
```
[the remark this answers is not in the document]
Comment (vishutrilio) [reply cmd/target-browser/cmd/constant.go]: Done

File: cmd/target-browser/cmd/resource-metadata.go [hunk 11]
```diff
@@ -15,7 +15,7 @@ func init() {
 func resourceMetadataCmd() *cobra.Command {
 	var cmd = &cobra.Command{
 		Use:   ResourceMetadataCmdName,
-		Short: "Get Backup high level metadata details",
+		Short: "Get Details of particular resource of a backup",
```
[the remark this answers is not in the document]
Comment (vishutrilio) [reply cmd/target-browser/cmd/resource-metadata.go]: Done

**Review by msurudwad — APPROVED** [review]
LGTM
